### PR TITLE
Bug 2035439: Use cloud egress network config for verifying egress IP in cloud environment

### PR DESCRIPTION
**Pull request by kyrtapz**
Since GCP uses a [`/32` mask](https://github.com/openshift/enhancements/blob/master/enhancements/network/cloud-egress-ip.md#gcp) it was not possible to assign an egress IP with the current SDN logic:
```E0104 17:55:57.694160    2014 egressip.go:143] Error assigning Egress IP "10.0.128.101": egress IP "10.0.128.101" is not in local network 10.0.128.2/32 of interface ens4```

This change introduces the usage of cloud egress network config to verify if the assignment is valid.

Signed-off-by: Patryk Diak <pdiak@redhat.com>

### CODE REVIEW

**Comment by kyrtapz** [comment]
@danwinship PTAL

File: pkg/network/node/egressip.go [hunk 1]
```diff
@@ -279,6 +280,29 @@ func (eip *egressIPWatcher) SetNamespaceEgressViaEgressIPs(vnid uint32, activeEg
...
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("unable to get egress link(%s) details: %v", eip.localIP, err)
+	}
+
+	if !eip.tracker.CloudEgressIP {
+		return localEgressLink, localEgressNet, nil, err
+	}
+
+	nodeName := eip.tracker.GetNodeNameByNodeIP(eip.localIP)
+	egresIPConfig, err := eip.tracker.GetNodeCloudEgressIPConfig(nodeName)
```
Comment (danwinship) [inline pkg/network/node/egressip.go:294]:
```suggestion
	egressIPConfig, err := eip.tracker.GetNodeCloudEgressIPConfig(nodeName)
```

**Review by danwinship — APPROVED** [review]
/approve
good except for the typo

**Review by alexanderConstantinescu — APPROVED** [review]
/bugzilla refresh
/lgtm

**Pull request merged**